### DISCUSSION TITLE
Keyboard Inputs never sent to dialog until content clicked on Unix based platforms

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#15295] [Q&A]
Avalonia Version: 11.0.10.

Currently having an issue where keyboard inputs aren't detected by the dialog window on Linux & macOS after being shown, activated & even focused using `.Focus()`.

I'm currently experiencing this issue on my Raspberry pi running the latest version of 64 bits RaspiOS & user also reported it to me happening on another linux distro, running KDE.

https://github.com/AvaloniaUI/Avalonia/assets/39861216/e929bdd7-c656-45d4-812e-0821806a9a24

Here is the code I'm currently using in the dialog

```cs
public partial class BindingEditorDialog : Window
{
    public BindingEditorDialog()
    {
        InitializeComponent();
    }

    protected override void OnDataContextChanged(EventArgs e)
    {
        base.OnDataContextChanged(e);

        if (DataContext is BindingEditorDialogViewModel vm)
        {
            // removed returned data
            vm.CloseRequested += (s, e) => Close("");
        }
    }

    protected override void OnKeyDown(KeyEventArgs e)
    {
        if (DataContext is BindingEditorDialogViewModel vm)
        {
            if (e.Key == Key.Escape)
            {
                Close(null);
            }
            else
            {
                // removed returned data
                Close("");
            }
        }

        base.OnKeyDown(e);
    }

    protected override void OnPointerPressed(PointerPressedEventArgs e)
    {
        // removed returned data
        if (DataContext is BindingEditorDialogViewModel vm)
            Close("");

        base.OnPointerPressed(e);
    }

    private string ParseMouseClick(PointerPressedEventArgs e)
    {
        var properties = e.GetCurrentPoint(this).Properties;

        if (properties.IsLeftButtonPressed)
            return "Left";
        else if (properties.IsRightButtonPressed)
            return "Right";
        else if (properties.IsMiddleButtonPressed)
            return "Middle";
        else
            return "None";
    }
}
```

I'm also noticing the window focus going back and forth between the main window & the opened dialog until one or the other is clicked. 
Clicking the title bar of the dialog window does not make it receive keyboard input until any of its content is clicked, at which point, the window will close in my case.
This works as intended on Windows.

Here is the link to a repo, where a control catalog is contained, The first page contains the control that open this dialog.
https://github.com/Mrcubix/OpenTabletDriver.External/

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/aab32860-192c-4fe4-bd51-1f95c8dbaecf)
![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/f17cf5c8-e115-4fac-8c5a-5057c7447765)

**Comment by timunie** [comment]
How do you set the Focus()?

**Reply by Mrcubix** [reply]
When Activated or loaded, by calling it on the window, or the content inside it.
EDIT: i just realized that both Focus calls fails, which is abnormal, & still doesn't work if doing it when it's attached to the visual tree.

~~I could also try to fetch the focus manager~~, but afaik, i shouldn't have to do anything in the first place.

EDIT: here is a clip of the current situation

https://github.com/AvaloniaUI/Avalonia/assets/39861216/e929bdd7-c656-45d4-812e-0821806a9a24

**Reply by Mrcubix** [reply]
Will probably transfer this to an issue by tomorrow, this is definitely a bug, the behavior should be the same no matter the platform.
Also not getting suggestions on workarounds, which will be faster to get through an issue.

**Reply by kovrom** [reply]
Strange, ControName.AttachedToVisualTree += (s, e) => ControlName.Focus();
usually works fine for me on linux

**Reply by Mrcubix** [reply]
Even if it did, i shouldn't have to do that (since it works as intended on windows with the same code) & the focus flicker is weird
